lesson_03/home_task_queries.sql

```diff
@@ -0,0 +1,89 @@
+/*
+ Завдання на SQL до лекції 03.
+ */
+
+
+/*
+1.
+Вивести кількість фільмів в кожній категорії.
+Результат відсортувати за спаданням.
+*/
+-- SQL code goes here...
+SELECT category.name     as category_name,
+       count(fc.film_id) as films_count
+FROM category
+         LEFT JOIN public.film_category fc on category.category_id = fc.category_id
+GROUP BY category.name
+ORDER BY films_count DESC;
+
+/*
+2.
+Вивести 10 акторів, чиї фільми брали на прокат найбільше.
+Результат відсортувати за спаданням.
+*/
+-- SQL code goes here...
+SELECT a.last_name, COUNT(r.rental_id) AS rental_count
+FROM actor a
+         LEFT JOIN film_actor fa ON a.actor_id = fa.actor_id
+         LEFT JOIN film f ON fa.film_id = f.film_id
+         LEFT JOIN inventory inv ON f.film_id = inv.film_id
+         LEFT JOIN rental r ON inv.inventory_id = r.inventory_id
+GROUP BY a.actor_id, a.first_name, a.last_name
+ORDER BY rental_count DESC
+LIMIT 10;
+
+/*
+3.
+Вивести категорія фільмів, на яку було витрачено найбільше грошей
+в прокаті
+*/
+-- SQL code goes here...
+WITH payments_by_categories AS (SELECT c.name AS category_name, SUM(p.amount) AS sum_payments
+                                FROM category c
+                                         JOIN film_category fc ON c.category_id = fc.category_id
+                                         JOIN inventory i ON i.film_id = fc.film_id
+                                         JOIN rental r ON r.inventory_id = i.inventory_id
+                                         JOIN payment p ON p.rental_id = r.rental_id
+                                GROUP BY c.name),
+     max_payment AS (SELECT MAX(sum_payments) AS max_sum_payments
+                     FROM payments_by_categories)
+SELECT category_name, sum_payments
+FROM payments_by_categories
+         JOIN max_payment ON payments_by_categories.sum_payments = max_payment.max_sum_payments;
```

Comment:
please format it like here:
<img width="596" alt="image" src="https://github.com/user-attachments/assets/eb6f1cf2-e9d6-4452-aa74-0bbdd2581061">

Reply:
Done